app/models/orders_package.rb

```diff
@@ -20,4 +22,95 @@ def set_initial_state
       transition :requested => :designated
     end
   end
+
+  def self.update_partially_designated_item(package)
+    orders_package = OrdersPackage.find(package[:orders_package_id].to_i)
```

Comment:
Instead of `find` it is better to to use find_by_id or where with first. Find will break if record not foud.